Server SIGSEGV for sideload commands

**Describe the bug**
SIGSEGV crash on sliver-server_linux for any `sideload` commands.

**To Reproduce**
Steps to reproduce the behavior:
1. Use a Windows beacon on a Linux Sliver Server.
2. Issue `sideload` for a DLL file.
3. Server crashes:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x90 pc=0xdb481d]

goroutine 570 [running]:
github.com/bishopfox/sliver/server/rpc.(*Server).Sideload(0x14e4479?, {0xc?, 0x14dd610?}, 0xc00090a100)
	github.com/bishopfox/sliver/server/rpc/rpc-tasks.go:173 +0x1bd
github.com/bishopfox/sliver/protobuf/rpcpb._SliverRPC_Sideload_Handler.func1({0xb638ca0, 0xc000768180}, {0x14451c0?, 0xc00090a100})
	github.com/bishopfox/sliver/protobuf/rpcpb/services_grpc.pb.go:3244 +0x7b
github.com/grpc-ecosystem/go-grpc-middleware/logging/logrus.PayloadUnaryServerInterceptor.func1({0xb638ca0, 0xc000768180}, {0x14451c0, 0xc00090a100}, 0x0?, 0xc0009860c0)
	github.com/grpc-ecosystem/go-grpc-middleware@v1.2.2/logging/logrus/payload_interceptors.go:29 +0x17d
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0xb638ca0?, 0xc000768180?}, {0x14451c0?, 0xc00090a100?})
	github.com/grpc-ecosystem/go-grpc-middleware@v1.2.2/chain.go:25 +0x3a
github.com/grpc-ecosystem/go-grpc-middleware/logging/logrus.UnaryServerInterceptor.func1({0xb638ca0, 0xc000768090}, {0x14451c0, 0xc00090a100}, 0xc0004a7ec0, 0xc0004a7ee0)
	github.com/grpc-ecosystem/go-grpc-middleware@v1.2.2/logging/logrus/server_interceptors.go:31 +0xe5
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0xb638ca0?, 0xc000768090?}, {0x14451c0?, 0xc00090a100?})
	github.com/grpc-ecosystem/go-grpc-middleware@v1.2.2/chain.go:25 +0x3a
github.com/grpc-ecosystem/go-grpc-middleware/tags.UnaryServerInterceptor.func1({0xb638ca0?, 0xc00037d9b0?}, {0x14451c0, 0xc00090a100}, 0xc0004a7ec0, 0xc0004a7f20)
	github.com/grpc-ecosystem/go-grpc-middleware@v1.2.2/tags/interceptors.go:23 +0xa6
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0xb638ca0?, 0xc00037d9b0?}, {0x14451c0?, 0xc00090a100?})
	github.com/grpc-ecosystem/go-grpc-middleware@v1.2.2/chain.go:25 +0x3a
github.com/bishopfox/sliver/server/transport.auditLogUnaryServerInterceptor.func1({0xb638ca0, 0xc00037d9b0}, {0x14451c0, 0xc00090a100}, 0xc0004a7ec0, 0xc0004a7f60)
	github.com/bishopfox/sliver/server/transport/middleware.go:208 +0x202
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0xb638ca0?, 0xc00037d9b0?}, {0x14451c0?, 0xc00090a100?})
	github.com/grpc-ecosystem/go-grpc-middleware@v1.2.2/chain.go:25 +0x3a
github.com/grpc-ecosystem/go-grpc-middleware/auth.UnaryServerInterceptor.func1({0xb638ca0, 0xc00037d5f0}, {0x14451c0, 0xc00090a100}, 0xc0004a7ec0, 0xc0004a7fa0)
	github.com/grpc-ecosystem/go-grpc-middleware@v1.2.2/auth/auth.go:47 +0xba
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0xb638ca0?, 0xc00037d5f0?}, {0x14451c0?, 0xc00090a100?})
	github.com/grpc-ecosystem/go-grpc-middleware@v1.2.2/chain.go:25 +0x3a
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1({0xb638ca0, 0xc00037d5f0}, {0x14451c0, 0xc00090a100}, 0xc000bdbae0?, 0x133db60?)
	github.com/grpc-ecosystem/go-grpc-middleware@v1.2.2/chain.go:34 +0xbf
github.com/bishopfox/sliver/protobuf/rpcpb._SliverRPC_Sideload_Handler({0x14b4580?, 0xbf890a8}, {0xb638ca0, 0xc00037d5f0}, 0xc00067a840, 0xc0003325a0)
	github.com/bishopfox/sliver/protobuf/rpcpb/services_grpc.pb.go:3246 +0x138
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00030d180, {0xb63fff8, 0xc0001ac000}, 0xc0007590e0, 0xc000332720, 0xbebd0e0, 0x0)
	google.golang.org/grpc@v1.42.0-dev.0.20211020220737-f00baa6c3c84/server.go:1282 +0xccf
google.golang.org/grpc.(*Server).handleStream(0xc00030d180, {0xb63fff8, 0xc0001ac000}, 0xc0007590e0, 0x0)
	google.golang.org/grpc@v1.42.0-dev.0.20211020220737-f00baa6c3c84/server.go:1616 +0xa1b
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.42.0-dev.0.20211020220737-f00baa6c3c84/server.go:921 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.42.0-dev.0.20211020220737-f00baa6c3c84/server.go:919 +0x28a
```

**Expected behavior**
Sideload the DLL in the remote process.

**Desktop**
 - Server OS: ArchLinux, go version go1.18.4 linux/amd64, 
 - Sliver version: v1.5.21 - fee6ee8f2d16f247481a61dfc817f0a52bf35f4c
 - Beacon runs on Windows 10 21H2

--- a/server/rpc/rpc-tasks.go
+++ b/server/rpc/rpc-tasks.go
@@ -150,14 +150,18 @@ func (rpc *Server) ExecuteAssembly(ctx context.Context, req *sliverpb.ExecuteAss
 
 // Sideload - Sideload a DLL on the remote system (Windows only)
 func (rpc *Server) Sideload(ctx context.Context, req *sliverpb.SideloadReq) (*sliverpb.Sideload, error) {
-	var session *core.Session
-	var beacon *models.Beacon
-	var err error
+	var (
+		session *core.Session
+		beacon  *models.Beacon
+		err     error
+		arch    string
+	)
 	if !req.Request.Async {
 		session = core.Sessions.Get(req.Request.SessionID)
 		if session == nil {
 			return nil, ErrInvalidSessionID
 		}
+		arch = session.Arch
 	} else {
 		beacon, err = db.BeaconByID(req.Request.BeaconID)
 		if err != nil {
@@ -167,10 +171,11 @@ func (rpc *Server) Sideload(ctx context.Context, req *sliverpb.SideloadReq) (*sl
 		if beacon == nil {
 			return nil, ErrInvalidBeaconID
 		}
+		arch = beacon.Arch
 	}
 
 	if getOS(session, beacon) == "windows" {
-		shellcode, err := generate.DonutShellcodeFromPE(req.Data, session.Arch, false, req.Args, "", "", req.IsDLL)
+		shellcode, err := generate.DonutShellcodeFromPE(req.Data, arch, false, req.Args, "", "", req.IsDLL)
 		if err != nil {
 			tasksLog.Errorf("Sideload failed: %s", err)
 			return nil, err

